File: System/Protocol/Allors.Protocol.Json/Api/Push/PushRequest.cs
// <copyright file="PushRequest.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Json.Api.Push
{
    using System.Text.Json.Serialization;

    public class PushRequest
    {
        [JsonPropertyName("n")]
        public PushRequestNewObject[] NewObjects { get; set; }

        [JsonPropertyName("o")]
        public PushRequestObject[] Objects { get; set; }
    }
}

File: typescript/modules/libs/protocol/json/system/src/lib/api/push/push-request.ts
import { Request } from '../request';
import { PushRequestNewObject } from './push-request-new-object';
import { PushRequestObject } from './push-request-object';

export interface PushRequest extends Request {
  /** NewObjects*/
  n?: PushRequestNewObject[];

  /** Objects */
  o?: PushRequestObject[];
}
